integration-testing/tests/find-users.test.js

```diff
@@ -103,3 +105,83 @@ test('Find users by phone, and by phone and email', async () => {
     .query({query: queries.findUsers, variables: {emails: [ourEmail, theirEmail], phoneNumbers: [theirPhone]}})
     .then(({data: {findUsers}}) => expect(findUsers.items.sort(cmp)).toEqual([us, them].sort(cmp)))
 })
+
+test('Find Users sends cards to the users that were found', async () => {
+  const {
+    client: ourClient,
+    userId: ourUserId,
+    email: ourEmail,
+    username: ourUsername,
+  } = await loginCache.getCleanLogin()
+  const {client: otherClient, userId: otherUserId, email: otherEmail} = await loginCache.getCleanLogin()
+  const {
+    client: other1Client,
+    userId: other1UserId,
+    email: other1Email,
+    username: other1Username,
+  } = await loginCache.getCleanLogin()
+  const {client: other2Client, userId: other2UserId, email: other2Email} = await loginCache.getCleanLogin()
+  const randomEmail = `${uuidv4()}@real.app`
+  const other1 = {__typename: 'User', userId: other1UserId, username: other1Username}
+
+  // find One User
+  await ourClient
+    .query({query: queries.findUsers, variables: {emails: [other1Email, randomEmail]}})
+    .then(({data: {findUsers}}) => expect(findUsers.items).toEqual([other1]))
+
+  // check called user has card
+  const cardId = await other1Client.query({query: queries.self}).then(({data: {self}}) => {
```

Comment:
```suggestion
  await misc.sleep(2000)
  const cardId = await other1Client.query({query: queries.self}).then(({data: {self}}) => {
```

We need a sleep in between the call to `User.findUsers()` and the call to check if the user that was found has a card. This is because the card for the user that was found is generated asynchronously.

Reply:
resolved